ansible_runner/utils/streaming.py

```diff
@@ -64,5 +72,11 @@ def unstream_dir(stream, length, target_directory):
             for info in archive.infolist():
                 archive.extract(info.filename, path=target_directory)
                 out_path = os.path.join(target_directory, info.filename)
-                perm = info.external_attr >> 16
-                os.chmod(out_path, perm)
+                perms = info.external_attr >> 16
+                mode = stat.filemode(perms)
+                if mode[:1] == 'l':
+                    link = open(out_path).read()
```

Comment:
could OOM if a malicious user puts a bunch of content in the symlink data portion. There are probably other/better way to OOM the system than that. Besides, the project SCM should be scrutinized via review process "why has our git checkout time jumped from 30 seconds to 3 hours?"

Reply:
https://unix.stackexchange.com/questions/32795/what-is-the-maximum-allowed-filename-and-folder-size-with-ecryptfs

>Linux has a maximum filename length of 255 characters for most filesystems (including EXT4), and a maximum path of 4096 characters.